polars/polars-lazy/polars-plan/src/logical_plan/optimizer/projection_pushdown/mod.rs

```diff
@@ -226,16 +226,18 @@ impl ProjectionPushDown {
         let root_projections = aexpr_to_leaf_nodes(proj, expr_arena);
 
         for (name, root_projection) in names.into_iter().zip(root_projections) {
-            let was_not_in_left = names_left.insert(name.clone());
-            let was_not_in_right = names_right.insert(name.clone());
```

Comment:
This inserted the name even if the column was not projected. 

That way a `filter` deeper down in the query with not add its column to the projection accumulations if the name was the hash table (incorrectly)